spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/DefaultMessagePropertiesConverter.java

```diff
@@ -155,11 +156,7 @@ else if (MessageProperties.RETRY_COUNT.equals(key)) {
 	@Override
 	public BasicProperties fromMessageProperties(final MessageProperties source, final String charset) {
 		BasicProperties.Builder target = new BasicProperties.Builder();
-		Map<String, Object> headers = convertHeadersIfNecessary(source.getHeaders());
-		long retryCount = source.getRetryCount();
-		if (retryCount > 0) {
-			headers.put(MessageProperties.RETRY_COUNT, retryCount);
-		}
```

Comment:
An alternative solution could be to keep the `retryCount` code here, but add some kind of check inside the `if (retryCount > 0)` block.

We could e.g., check if `headers` is empty:
```java
...
if (retryCount > 0) {
	if (headers.isEmpty()) {
		headers = new HashMap<>();
	}
	headers.put(MessageProperties.RETRY_COUNT, retryCount);
}
...
```
or catch the exception:
```java
...
if (retryCount > 0) {
	try {
		headers.put(MessageProperties.RETRY_COUNT, retryCount);
	} catch (UnsupportedOperationException e) {
		headers = Map.of(MessageProperties.RETRY_COUNT, retryCount);
	}
}
...
```